management/tests/commands/add-identity-to-user.test.ts

```diff
@@ -0,0 +1,77 @@
+import { Server } from "http";
+
+import { addIdentityToUser } from "../../src/commands";
+import { InvalidValidationCodeError } from "../../src/errors";
+import * as checkVerificationCode from "../../src/queries/check-verification-code";
+import { getIdentityType } from "../../src/utils/get-identity-type";
+import { nonPrimaryNewIdentity } from "../mocks/identities";
+import { app } from "../mocks/test-server/server";
+
+describe("Add identity to user", () => {
+  let server: Server;
+
+  beforeAll(async () => {
+    await new Promise(
+      (resolve) => (server = app.listen(3000, resolve as () => void)),
+    );
+  });
```

Comment:
```suggestion
  beforeAll(async () => {
    await new Promise(
      (resolve) => (server = app.listen(3000, () => resolve())),
    );
  });
```

This should avoid the `as`, doesn't it?

Reply:
I will try it 🙂

Reply:
```typescript
beforeAll(async () => {
    await new Promise<void>(
      (resolve) => (server = app.listen(3000, () => resolve())),
    );
});
```
So I tried with that, and I'm having an issue where the server doesn't seems to close correctly between two tests run. I can't figure out why 😕

Reply:
is `server` not assigned properly?

Reply:
What do you mean ?

Reply:
Well, it's weird that changing how `beforeAll` works lead to something not closing in the end.
That's why my guess goes to the global variable `server` not being assigned properly.

Which we could test by console logging `server` in the `afterAll` to see if it's undefined or not.

Reply:
Well, I logged `server` in both `afterAll` and they are defined in both case. I still have ` listen EADDRINUSE: address already in use :::3000` when trying to launch the server in the second test running.

Reply:
ok let me try your branch

Reply:
Ok, I can't explain it at the moment but ...

```diff
   beforeAll(async () => {
     await new Promise<void>(
-      (resolve) => (server = app.listen(3000, () => resolve())),
+      (resolve) => {server = app.listen(3000, () => resolve())},
     );
   });
```

Changing parentheses to curly braces makes it work 😅

Reply:
Alright, something is definitely very strange here. It still doesn't work when trying to run the tests in local, BUT I just pushed this change and it worked in CI. 🤯

Reply:
if you still had a server running because of the previous tests, it's logical it failed locally. Maybe after a reboot? 😅

Reply:
But in this case, the first test run using the server won't pass either, no ? Because there, only the one executed in second fails with the already-in-use error locally 🤔

Reply:
uuuuh 🤯